src/proxy/http/mod.rs

```diff
@@ -29,12 +29,25 @@ pub trait HasH2Reason {
     fn h2_reason(&self) -> Option<::h2::Reason>;
 }
 
+impl HasH2Reason for Box<dyn std::error::Error> {
+    fn h2_reason(&self) -> Option<::h2::Reason> {
+        None
```

Comment:
Should this check `self.source` to see if it implements HasH2Reason? is that even possible?

Reply:
Woah, good catch! I meant to fill this in, shoulda left myself a better note.